engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java

```diff
@@ -345,6 +353,35 @@ public Optional<Module> findModuleByExpressionId(UUID expressionId) {
         .findFirst();
   }
 
+  /**
+   * Tries to lookup a Java class (host symbol in Truffle terminology) by its fully qualified name.
+   * This method also tries to lookup inner classes. More specifically, if the provided name
+   * resolves to an inner class, then the import of the outer class is resolved, and the inner class
+   * is looked up by iterating the members of the outer class via Truffle's interop protocol.
+   *
+   * @param className Fully qualified class name, can also be nested static inner class.
+   * @return The java class or Optional.empty if not found.
+   */
+  public Optional<Object> lookupJavaClass(String className) {
```

Comment:
Wish: no `Optional` please!

Reply:
Removed in a7d41ffff96c25c8dd5c529809b9f2ba3f45054b - I wanted to be consistent with the rest of the public methods in `EnsoContext`. But I prefer standard `null` usage in Java to `Optional` as well.